src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/NuGetTelemetryProvider.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using NuGet.Common;
+
+namespace NuGet.VisualStudio.Telemetry
+{
+    [Export(typeof(INuGetTelemetryProvider))]
+    internal class NuGetTelemetryProvider : INuGetTelemetryProvider
+    {
+        public Task EmitEventAsync(TelemetryEvent telemetryEvent)
+        {
+            TelemetryActivity.EmitTelemetryEvent(telemetryEvent);
+            return Task.CompletedTask;
+        }
+
+        public async Task PostFaultAsync(Exception e, string callerClassName, [CallerMemberName] string callerMemberName = null, IDictionary<string, object> extraProperties = null)
+        {
+            await TelemetryUtility.PostFaultAsync(e, callerClassName, callerMemberName, extraProperties);
+        }
+
+        public void PostFault(Exception e, string callerClassName, [CallerMemberName] string callerMemberName = null, IDictionary<string, object> extraProperties = null)
+        {
+            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
```

Comment:
do you need the async + await?

Reply:
no, but David Fowler seems to be suggesting to do this anyway: https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md#prefer-asyncawait-over-directly-returning-task

Reply:
I'll leave it up to you then :) 

I'll get nerd sniped into understanding what in particular the benefits might/might not be, and that's a task best left for another day.